src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -391,6 +400,115 @@ public void createMissingClass(UnsolvedClass missedClass) {
+  /**
+   * This method checks if a string is capitalized
+   *
+   * @param string the string to be checked
+   * @return true if the string is capitalized
+   */
+  public static boolean isCapital(String string) {
+    String capitalizedString = toCapital(string);
+    return string.equals(capitalizedString);
+  }
+
+  /**
+   * This method checks if a string is a class path.
+   *
+   * @param potentialClassPath the string to be checked
+   * @return true if the string is a class path
+   */
+  public static boolean isAClassPath(String potentialClassPath) {
+    String[] elements = potentialClassPath.split(".");
+    int elementsCount = elements.length;
+    return elementsCount > 1 && isCapital(elements[elementsCount - 1]);
+  }
+
+  /**
+   * Given a class path, this method will create a simple instance of UnsolvedClass for that class
+   * path
+   *
+   * @param classPath the class path
+   * @return the corresponding instance of UnsolvedClass
+   */
+  public static UnsolvedClass getSimpleSyntheticClassFromClassPath(String classPath) {
+    if (!isAClassPath(classPath)) {
+      throw new RuntimeException(
+          "Check with isAClassPath first before using getSimpleSyntheticClassFromClassPath");
+    }
+    String[] elements = classPath.split(".");
+    // the last element of a class path is the simple name of that class
+    @SuppressWarnings("signature")
+    @ClassGetSimpleName String className = elements[elements.length - 1];
+    String packageName = classPath.replace("." + className, "");
+    return new UnsolvedClass(className, packageName);
+  }
+
+  /**
+   * This method checks if a method call is not-simple and unsolved. In this context, we declare a
+   * not-simple method call as a method that is directly called by a qualified class name. For
+   * example, for this call org.package.Class.methodFirst().methodSecond(),
+   * "org.package.Class.methodFirst()" is a not-simple method call, but
+   * "org.package.Class.methodFirst().methodSecond()" is a simple one.
+   *
+   * @param method the method call to be checked
+   * @return true if the method call is not simple and unsolved
+   */
+  public static boolean unsolvedAndNotSimple(MethodCallExpr method) {
+    try {
+      method.resolve().getReturnType();
+      return false;
+    } catch (Exception e) {
+      Optional<Expression> callerExpression = method.getScope();
+      if (callerExpression.isEmpty()) {
+        return false;
+      }
+      String callerToString = callerExpression.get().toString();
+      return isAClassPath(callerToString);
+    }
+  }
+
+  /**
+   * For a method call that is not simple, this method will take that method as input and create corresponding synthetic class
+   * @param methodCall the method call to be taken as input
+   */
+  public void updateClassSetWithNotSimpleMethodCall(String methodCall) {
+    String methodCallWithoutParen = methodCall.replace("()", "");
+    String[] methodParts = methodCallWithoutParen.split("[^a-zA-Z]+");
```

Comment:
Method names can include numbers (e.g., `test1()`, `test2()`, etc.). Can you add a test case with a number in the name of a method that is called in a non-simple method call and make sure that this code handles it properly?

We may also need to handle non-ascii characters. Here's the relevant part of the JLS: https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8 (note that most unicode are `JavaLetter`s, and therefore valid in names, so we need to be able to handle them).